Tasks/Maven/CodeAnalysis/mavensonar.ts

```diff
@@ -0,0 +1,112 @@
+        var sqDbPassword = tl.getInput('sqDbPassword', false);
+        mvnsq = createMavenSonarQubeRunner(mvnPath, sqEndpoint.Url, sqEndpoint.Username, sqEndpoint.Password, sqDbUrl, sqDbUsername, sqDbPassword);
+    }
+    else {
+        mvnsq = createMavenSonarQubeRunner(mvnPath, sqEndpoint.Url, sqEndpoint.Username, sqEndpoint.Password);
+    }
+
+    // Apply argument for the JaCoCo tool, if enabled
+    if (typeof execFileJacoco != "undefined" && execFileJacoco) {
+        mvnsq.arg('-Dsonar.jacoco.reportPath=' + execFileJacoco);
+    }
+
+    mvnsq.arg('-f');
+    mvnsq.pathArg(mavenPOMFile);
+    mvnsq.argString(mavenOptions); // add the user options to allow further customization of the SQ run
+    mvnsq = sqCommon.applySonarQubeIssuesModeInPrBuild(mvnsq); // in PR builds run SQ in issues mode
+    mvnsq.arg("sonar:sonar");
+
+    return mvnsq;
+}
+
+// Upload a build summary with links to available SonarQube dashboards for further analysis details.
+export function uploadSonarQubeBuildSummary(): void {
+    // Collect all unique dashboard URLs
+    // project key -> dashboard URL
+    var dashboardUrlsByProjectKey: Map<string, string> = new Map<string, string>();
+
+    var sourcesDir = tl.getVariable('build.sourcesDirectory');
+    var modules:ModuleAnalysis[] = codeAnalysis.findCandidateModules(sourcesDir);
+    modules.forEach((module: ModuleAnalysis) => {
+        var taskReport = sqCommon.getSonarQubeTaskReport(path.join(module.rootDirectory, 'target', 'sonar'));
+        if (taskReport) {
+            dashboardUrlsByProjectKey.set(taskReport['projectKey'], taskReport['dashboardUrl']);
+        }
+    });
+
+    dashboardUrlsByProjectKey.delete(null);
+
+    if (dashboardUrlsByProjectKey.size < 1) {
+        tl.debug('Expected to find SonarQube report file(s) on disk, but found none.');
+        return;
+    }
+
+    // Create a build summary line-by-line, write it out and upload
+    var buildSummaryLines:string[] = [];
+    var lineBodyText:string;
+
+   for (let projectKey of dashboardUrlsByProjectKey.keys()) {
+       // If there is more than one project key, we will append it to the build summary line text
+       if (dashboardUrlsByProjectKey.size > 1) {
+           // Looks like: "Detailed SonarQube report (foo)"
+           lineBodyText = tl.loc('sqAnalysisBuildSummaryLine_SomeProjects', projectKey);
+       } else {
+           // Looks like: "Detailed SonarQube report"
+           lineBodyText = tl.loc('sqAnalysisBuildSummaryLine_OneProject');
+       }
+
+       buildSummaryLines.push(util.format('[%s >](%s "%s Dashboard")',
+           lineBodyText, dashboardUrlsByProjectKey.get(projectKey), projectKey));
+   };
+
+    // Save and upload build summary
+    // Looks like: "[Detailed SonarQube report (foo) >](https://mySQserver:9000/dashboard/index/foo "foo Dashboard")  \r\n
+    // [Detailed SonarQube report (bar) >](https://mySQserver:9000/dashboard/index/bar "bar Dashboard")  \r\n"
```

Comment:
Function now has the following sections:

- Parse
- Create summary
- Save to disk
- Upload

Resolved, done.